### ISSUE TITLE
ArchiveExtractor.extractArchive() skips files specified with -d option, only dirs are processed

### ISSUE THREAD

**Issue by justincranford** [whitesource/fs-agent#8]
CommandLineArgs.java:39 says files and directories are supported by the -d parameter.

```
    @Parameter(names = "-d", splitter = CommaParameterSplitter.class, description = "Comma separated list of directories and / or files to scan")
```

I tried to specify a tar file in the -d parameter, but it is ignored. I debugged through 1.8.5 tag code and I think there is a missing "else if" check in ArchiveExtractor.extractArchive(). At ArchiveExtractor.java:175, the -d parameter list is checked to see if it is a directory, but there is no corresponding logic to handle a value that is a file instead of a directory.

Would something like this would work better? The archive file extraction can be moved to a helper, so extractArchive() can check for file or directory and handle both.

```
    private void extractArchive(String scannerBaseDir, String destDirectory, int archiveExtractionDepth, int curLevel) {
        File file = new File(scannerBaseDir);
        if (file.exists()) {
            if (file.isFile()) {
                extractArchiveHelper(scannerBaseDir, destDirectory, archiveExtractionDepth, curLevel, scannerBaseDir);
            } else if (file.isDirectory()) {
                // scan directory
                DirectoryScanner scanner = new DirectoryScanner();
                scanner.setBasedir(scannerBaseDir);
                scanner.setIncludes(archiveIncludesPattern);
                scanner.setExcludes(archiveExcludesPattern);
                scanner.setCaseSensitive(false);
                scanner.scan();

                String[] fileNames = scanner.getIncludedFiles();
                for (String fileName : fileNames) {
                    extractArchiveHelper(scannerBaseDir, destDirectory, archiveExtractionDepth, curLevel, fileName);
                }
            }
        }
    }
```

```

	private void extractArchiveHelper(String scannerBaseDir, String destDirectory, int archiveExtractionDepth, int curLevel, String fileName) {
		String innerDir = destDirectory + File.separator + fileName + RANDOM_STRING;
		String archiveFile = scannerBaseDir + File.separator + fileName;
		String lowerCaseFileName = fileName.toLowerCase();
		if (lowerCaseFileName.matches(ZIP_EXTENSION_PATTERN)) {
		    unZip(lowerCaseFileName, innerDir, archiveFile);
		} else if (lowerCaseFileName.matches(GEM_EXTENSION_PATTERN)) {
		    unTar(lowerCaseFileName, innerDir, archiveFile);
		    innerDir = innerDir + File.separator + RUBY_DATA_FILE;
		    unTar(RUBY_DATA_FILE, innerDir + RANDOM_STRING, innerDir);
		    innerDir = innerDir + RANDOM_STRING;
		} else if (lowerCaseFileName.matches(TAR_EXTENSION_PATTERN)) {
		    unTar(lowerCaseFileName, innerDir, archiveFile);
//                        innerDir = innerDir.replaceAll(TAR_SUFFIX, BLANK);
		} else if (lowerCaseFileName.matches(RPM_EXTENSION_PATTERN)) {
		    handleRpmFile(fileName, innerDir, archiveFile);
		} else if (lowerCaseFileName.matches(RAR_EXTENSION_PATTERN)) {
		    File destDir = new File(innerDir);
		    if (!destDir.exists()) {
		        destDir.mkdirs();
		    }
		    ExtractArchive.extractArchive(archiveFile, innerDir);
		} else {
		    logger.warn("Error: {} is unsupported archive type", fileName);
		    return;
		}
		// Extract again if needed according archiveExtractionDepth parameter
		if (curLevel < archiveExtractionDepth) {
		    extractArchive(innerDir, innerDir, archiveExtractionDepth, curLevel + 1);
		}
	}
```

**Comment by tomshapira** [comment]
Resolved in commit [99c714b](https://github.com/whitesource/fs-agent/commit/99c714b2889d86f68c8def689915a8a7db3e6ff5): support scanning single archive files using -d parameter

**Comment by justincranford** [comment]
I confirmed this was fixed in 1.8.7.
